### PR TITLE
Add fullscreen mode to responsive controls

### PR DESCRIPTION
**Pull request by halocline**
[Deploy Preview](https://deploy-preview-2240--keen-mayer-a86c8b.netlify.app/templates/page-layouts)

#### What does this PR do?

Preparatory work to make view responsive behavior calls to action more prominent.

#### Where should the reviewer start?

ResponsiveControls.js
Example.js

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### How should this be manually tested?

Modify an `<Example />` such as on `page-layouts.mdx` and test various configurations for the `showResponsiveControls` prop.

Should show default Desktop, Laptop, Mobile controls:
```
<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/page-layouts/HeaderFooterExample.js"
  figma="https://www.figma.com/file/4rdBkUlHd5MCVq3hvUOXHd/App-Page-Layouts?node-id=90%3A6545"
  relevantComponents={['Header', 'Footer', 'Main']}
  width="100%"
  screenContainer
>
  <HeaderFooterExample />
</Example>
```

Should show only controls contained in array:
```
<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/page-layouts/HeaderFooterExample.js"
  figma="https://www.figma.com/file/4rdBkUlHd5MCVq3hvUOXHd/App-Page-Layouts?node-id=90%3A6545"
  relevantComponents={['Header', 'Footer', 'Main']}
  width="100%"
  screenContainer
  showResponsiveControls={['fullScreen', 'mobile']}
>
  <HeaderFooterExample />
</Example>
```

Should not show any controls:
```
<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/page-layouts/HeaderFooterExample.js"
  figma="https://www.figma.com/file/4rdBkUlHd5MCVq3hvUOXHd/App-Page-Layouts?node-id=90%3A6545"
  relevantComponents={['Header', 'Footer', 'Main']}
  width="100%"
  screenContainer
  showResponsiveControls={false}
>
  <HeaderFooterExample />
</Example>
```

### CODE REVIEW

**Comment by britt6612** [comment]
are we planning to keep the bottom expand seems like its repeating the above control?
Screen Shot 2022-01-21 at 10 23 17 AM

File: aries-site/src/layouts/content/Example/Example.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const Example = ({
   ...rest
 }) => {
   const [screen, setScreen] = React.useState(screens.laptop);
-  const [showLayer, setShowLayer] = React.useState(false);
+  const [fullscreen, setFullscreen] = React.useState(false);
```
Comment (taysea) [inline aries-site/src/layouts/content/Example/Example.js:51]: thank you

**Comment by taysea** [comment]
Code looks good, but agreed that having the fullscreen button in two places is a bit confusing

**Comment by halocline** [comment]
> are we planning to keep the bottom expand seems like its repeating the above control?

I debated this as well. I can go either way depending on thoughts of others.

My thoughts:
- For certain examples, demoing the fullscreen/responsive behaviors is something we want to encourage -- if not the primary thing we want readers to see (e.g. page layout examples).
  - I feel that the "Interact in Fullscreen" control does a nice job emphasizing the call to action when that is the desired behavior.
- For other examples, the see fullscreen is kind of a nice to have and doesn't need the prominence.
  - The current positioning of the expand is nice for the "nice to have" scenarios because it is more subtle.
  - The vast majority of examples on the site currently have this "nice to have" expand and we probably want to continue providing this functionality.
- We could:
  - Leave both when "Interact in Fullscreen" is present -- I don't know that the redundancy hurts the experience materially.
  - Remove the "nice to have" when "Interact in Fullscreen" is present.
  - Consider moving the "nice to have" to always appear where the ResponsiveControls live, but give it the option to be displayed with or without the "Interact in Fullscreen" label -- This would allow us to emphasize when desired, and present subtly when "nice to have".
    - If we headed in this direction, we could potentially repurpose the real-estate where the ResponsiveControls live, turning it into an "Example Controls" area. The "Example Controls" area could be a flexible area where actions specific to the example could be applied.
    - The "Example Controls" area could behave similarly to Storybook's "Controls" where various props could be applied.
    - The "Example Controls" area could present things like the current screen modes, fullscreen, themeMode toggle specific to the example, props specific to the example, "layers controls" (i.e. ability to toggle between hi-fidelity examples and more "boxy"/wireframe examples, etc.
    
While typing, I'm kind of liking the idea of repurposing the ResponsiveControls area to an apply properties to the example area...

**Comment by halocline** [comment]
Implemented the following based on our discussion:
- When `showResponsiveControls` show fullscreen control inline with the responsive controls and remove from the example footer.
- Some polishing such as animation, wrapping for narrow screens, use of `tip`, ...

File: aries-site/src/layouts/content/Example/Example.js [hunk 2]
```diff
@@ -226,19 +229,26 @@ export const Example = ({
                     controls={showResponsiveControls}
                     onSetScreen={setScreen}
                     screen={screen}
+                    fullscreen={fullscreen}
+                    setFullscreen={setFullscreen}
                   />
                 )}
                 <Button
-                  title="Leave full screen"
+                  tip="Leave Fullscreen"
```
Comment (taysea) [inline aries-site/src/layouts/content/Example/Example.js:237]: nice

File: aries-site/src/layouts/content/Example/Example.js [hunk 3]
```diff
@@ -226,19 +229,26 @@ export const Example = ({
...
                   onClick={() => {
-                    setShowLayer(false);
+                    setFullscreen(false);
                     setScreen(screens.laptop);
                   }}
                   hoverIndicator
                 />
               </Box>
               <Box
+                animation={{
```
Comment (taysea) [inline aries-site/src/layouts/content/Example/Example.js:247]: This animation felt a little slow to me -- I was wondering if the example wasn't loading -- can we reduce the time / delay a little?

Comment (halocline) [reply aries-site/src/layouts/content/Example/Example.js:247]: Done.

**Review by taysea — APPROVED** [review]
LGTM

File: aries-site/src/layouts/content/Example/Example.js (hunk 3 above)
Comment (britt6612) [reply aries-site/src/layouts/content/Example/Example.js:247]: looks nice

**Review by britt6612 — APPROVED** [review]
Im good with these changes

**Pull request merged**